### PR TITLE
feat: package file delete on package file object

### PR DESCRIPTION
**Pull request by simonisateur**

### CODE REVIEW

**Comment by JohnVillalovos** [comment]
Thanks. Would you be willing to add a test case for this? Probably in `tests/functional/api/test_packages.py` would be good.

**Comment by simonisateur** [comment]
@JohnVillalovos I've added a unit tests and classified the merge as a `feat`.

**Comment by simonisateur** [comment]
@JohnVillalovos I've fixed my test. I like the setup of the project with commitlint.

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -1,5 +1,11 @@
 name: Docs
 
+# If a pull-request is pushed then cancel all previously running jobs related
+# to that pull-request
```
Comment (JohnVillalovos) [inline .github/workflows/docs.yml:4]: This looks wrong.

I think the MR needs to be rebased.

**Review by JohnVillalovos — CHANGES_REQUESTED** [review]

File: .github/workflows/docs.yml (hunk 1 above)
Comment (simonisateur) [reply .github/workflows/docs.yml]: Yes it does. I fixed it

Comment (JohnVillalovos) [reply .github/workflows/docs.yml]: Thanks. Can you squash it down to one commit? Instead of the current commit and then a "merge commit of main branch"

Comment (simonisateur) [reply .github/workflows/docs.yml]: yes

Comment (JohnVillalovos) [reply .github/workflows/docs.yml]: > yes

Awesome! Thank you.

Comment (simonisateur) [reply .github/workflows/docs.yml]: All in one commit. My code and the merge

Comment (simonisateur) [reply .github/workflows/docs.yml]: Is the message `feat: add delete on package_file` object ok ?

**Comment by JohnVillalovos** [comment]
Thanks. As a note for the future. It can sometimes be simpler to do the tests in the `tests/functional/api/` directory as those actually run against a running Gitlab instance.

But thanks for the unit tests!

**Review by JohnVillalovos — APPROVED** [review]

**Comment by simonisateur** [comment]
Thanks for your precious time. I will note that! As a fact I will revisit the functional test in the future to help me learn pytest and be more confident in the delete at the same time.

My delete is'nt a great test because it only test if it does'nt work, but not if it works.

**Comment by nejch** [comment]
Thank you very much @simonisateur!

**Pull request merged**